RFS/src/main/java/com/rfs/RfsMigrateMetadata.java

```diff
@@ -0,0 +1,130 @@
+package com.rfs;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+
+
+import com.rfs.cms.CmsClient;
+import com.rfs.cms.OpenSearchCmsClient;
+import com.rfs.common.ClusterVersion;
+import com.rfs.common.ConnectionDetails;
+import com.rfs.common.GlobalMetadata;
+import com.rfs.common.IndexMetadata;
+import com.rfs.common.Logging;
+import com.rfs.common.OpenSearchClient;
+import com.rfs.common.S3Uri;
+import com.rfs.common.S3Repo;
+import com.rfs.common.SourceRepo;
+import com.rfs.common.TryHandlePhaseFailure;
+import com.rfs.common.SnapshotRepo;
+import com.rfs.transformers.TransformFunctions;
+import com.rfs.transformers.Transformer;
+import com.rfs.version_es_7_10.GlobalMetadataFactory_ES_7_10;
+import com.rfs.version_es_7_10.IndexMetadataFactory_ES_7_10;
+import com.rfs.version_es_7_10.SnapshotRepoProvider_ES_7_10;
+import com.rfs.version_os_2_11.GlobalMetadataCreator_OS_2_11;
+import com.rfs.version_os_2_11.IndexCreator_OS_2_11;
+import com.rfs.worker.GlobalState;
+import com.rfs.worker.IndexRunner;
+import com.rfs.worker.MetadataRunner;
+
+public class RfsMigrateMetadata {
+    private static final Logger logger = LogManager.getLogger(RfsMigrateMetadata.class);
+
+    public static class Args {
+        @Parameter(names = {"--snapshot-name"}, description = "The name of the snapshot to migrate", required = true)
```

Comment:
It would be nice to enforce this string/description (when possible) to be defined in one place, either via inheritance, if that works, or by pulling in string constants from some common file (probably safer and more explicit especially w/ more than 2-3 applications).  Eventually, those should harmonize w/ the proxy and replayer too, but no rush.